lib/src/io.cpp

```diff
@@ -303,6 +309,7 @@ record& stream::at( int i, record& rec ) noexcept (false) {
 
             if (err) consistent = false;
             attributes.push_back( attrs );
+            types.push_back( type );
```

Comment:
What's going on here?

Reply:
Every record has a "type" field. There is a line further in the method that reads type from types vector (types.front()). But we never put current type there, so it was always reading "0" from empty vector before.